<?xml version="1.0"?>
<launch>

    
<!--
    <node pkg="rosbag" type="play" name="rosbag_play" args="&#45;&#45;clock -r 0.5 $(env HOME)/Downloads/2011_09_30_0018.bag"/>
-->
    <!--- Sim Time -->
    <param name="/use_sim_time" value="true" />
    <param name="scan_period" value="0.1" />

    <param name="vertical_angle" type="double" value="2.0" />
    <param name="max_dis" type="double" value="150.0" />
    <param name="map_resolution" type="double" value="0.3" />
    <param name="min_dis" type="double" value="1.5" />

    <param name="num_sub_cloud" type="int" value="1" />
    <param name="lidar_type" type="int" value="1" />

    <param name="max_search_dis" type="double" value="1" />
    <param name="optimization_times" type="int" value="6" />
    <param name="iteration_times" type="int" value="8" />
    <param name="box_sides" type="double" value="100" />
    <param name="feature_resolution" type="double" value="0.4" />
    <!--- -->
    <node pkg="floam" type="floam_odom_estimation_node" name="floam_odom_estimation_node" output="screen"/>
    <node pkg="floam" type="floam_laser_processing_node" name="floam_laser_processing_node" output="screen"/>
    <node pkg="floam" type="floam_laser_mapping_node" name="floam_laser_mapping_node" output="screen"/>

    <node pkg="tf" type="static_transform_publisher" name="word2map_tf"  args="0 0 0 0 0 0 /world /map 10" />
    <node pkg="tf" type="static_transform_publisher" name="base2rslidar_tf"  args="0 0 0 0 0 0 /base_link /rslidar 10" />

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find floam)/rviz/floam_M1.rviz" />
    </group>

    
  	<node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="gt" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="rslidar" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>
    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="base_link" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="base_link" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>

</launch>
